### PR TITLE
reconciliation processor to make requests to down streams and save to db

### PR DESCRIPTION
**Pull request by mdayican**
### Change description ###

reconciliation processor to make requests to down streams and save to db

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/ReconciliationProcessService.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            switch (targetStorageAccount) {
+                case BULKSCAN:
+                    processBulkScan(reconciliationStatement, supplierStatement.id);
+                    break;
+                case CRIME:
+                    logger.error("Crime reconciliation process skipped");
+                    break;
+                case PCQ:
+                    logger.error("PCQ reconciliation processing skipped");
+                    break;
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/ReconciliationProcessService.java:77]: should the loggin level be 'error' here?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/ReconciliationProcessService.java:77]: what about high level reconciliation?

File: src/test/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/ReconciliationProcessServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+            reconciliationMapper,
+            reconciliationService,
+            bulkScanProcessorClient,
+            objectMapper,
+            reconciliationReportRepository
+        );
+    }
+
+    @Test
+    void should_only_interact_reconciliationService_when_no_supplier_statement() {
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/ReconciliationProcessServiceTest.java:64]: lets be more specific

File: src/test/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/ReconciliationProcessServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        verify(reconciliationService).getSupplierStatement(date);
+        verifyNoMoreInteractions(reconciliationService);
+        verifyNoInteractions(reconciliationMapper);
+        verifyNoInteractions(bulkScanProcessorClient);
+        verifyNoInteractions(objectMapper);
+        verifyNoInteractions(reconciliationReportRepository);
+    }
+
+    @Test
+    void should_throw_exception_when_getting_supplier_statement() {
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/ReconciliationProcessServiceTest.java:82]: looks like there's something missing at the end

File: src/test/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/ReconciliationProcessServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        assertThat(newReconciliationReport.contentTypeVersion).isEqualTo("1.0");
+        assertThat(newReconciliationReport.supplierStatementId).isEqualTo(supplierStatementId);
+
+        verifyNoMoreInteractions(reconciliationService);
+        verifyNoMoreInteractions(reconciliationMapper);
+        verifyNoMoreInteractions(bulkScanProcessorClient);
+        verifyNoMoreInteractions(objectMapper);
+        verifyNoMoreInteractions(reconciliationReportRepository);
+    }
+}
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/ReconciliationProcessServiceTest.java:187]: new line please

File: src/test/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/ReconciliationServiceTest.java [hunk 5]
```diff
@@ -102,4 +106,35 @@ void should_throw_exception_when_supplier_statement_json_processing_fails() thro
             .hasMessageContaining("Failed to process Supplier statement");
     }
 
+    @Test
+    void should_return_supplier_statement_as_repository() {
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/ReconciliationServiceTest.java:110]: definitely worth rephrasing

File: src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/ReconciliationProcessService.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+    public void process(LocalDate date) {
+        var optionalEnvelopeSupplierStatement = reconciliationService.getSupplierStatement(date);
+        if (optionalEnvelopeSupplierStatement.isPresent()) {
+            var supplierStatement = optionalEnvelopeSupplierStatement.get();
+            for (var targetStorage : TargetStorageAccount.values()) {
+                processByTargetStorage(supplierStatement, targetStorage);
+            }
+        } else {
+            logger.info("No Supplier Statement for date: {}", date);
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/ReconciliationProcessService.java:56]: that's quite a problem if there is none, perhaps an error so we get alerted?

**Comment by mdayican** [comment]
will recreate new one

**Pull request closed without merging**